devel/api.md

```diff
@@ -0,0 +1,1074 @@
+
+The names of bond slave interfaces.
+This property does not support partial editing, full list of salves is required
+in desired state.
+
+# Interface -- Open vSwitch(OVS) Bridge
+
+Besides basic interface properties, each OVS bridge interface state also
+contains a dictionary saved in key `OVSBridge.CONFIG_SUBTREE`.
+
+Example:
+
+```python
+
+{
+    Interface.KEY: [
+        Interface.NAME: "ovs-br0",
+        Interface.TYPE: InterfaceType.BOND,
+        Interface.STATE: InterfaceState.UP
+        OVSBridge.CONFIG_SUBTREE: {
+            OVSBridge.OPTIONS_SUBTREE: {
+                OVSBridge.FAIL_MODE: "",
+                OVSBridge.MCAST_SNOOPING_ENABLE: false,
+                OVSBridge.RSTP: false,
+                OVSBridge.STP: true
+            },
+            OVSBridge.PORT_SUBTREE: [
+                {
+                    OVSBridge.PORT_NAME: "ovs0",
+                    OVSBridge.PORT_TYPE: OVSBridgePortType.INTERNAL
+                },
+                {
+                    OVSBridge.PORT_NAME: "eth1",
+                    OVSBridge.PORT_TYPE: OVSBridgePortType.SYSTEM
+                }
+            ]
+        },
+    ]
+}
+```
+
+## `OVSBridge.FAIL_MODE`
+
+Type: `string`, 'secure' or 'standalone' or empty.
+
+The bridge failure mode.
+
+## `OVSBridge.MCAST_SNOOPING_ENABLE`
+
+Type: `bool`
+
+Enable or disable multicast snooping.
+
+## `OVSBridge.RSTP`
+
+Type: `bool`
+
+Enable or disable RSTP.
+
+## `OVSBridge.STP`
+
+Type: `bool`
+
+Enable or disable STP.
+
+## `OVSBridge.PORT_SUBTREE`
+
+Type: List of dictionary.
+
+This property does not support partial editing, the full list of ports is
+required in desired state.
+
+
+## `OVSBridge.PORT_NAME`
+
+Type: `string`
+
+The OVS bridge port name.
+
+## `OVSBridge.PORT_TYPE`
+
+Type: `string`, `OVSBridgePortType.SYSTEM` or `OVSBridgePortType.INTERNAL`
+
+The OVS bridge port type:
+ * `OVSBridgePortType.SYSTEM`
+   The interface attached to this OVS bridge.
+
+ * `OVSBridgePortType.INTERNAL`
+   The interface represents the bridge itself in kernel.
+   There is a interface created with `InterfaceType.OVS_INTERFACE` type
+   for internal OVS interface allowing IPv4/IPv6 configurations.
+
+# Route
+
+Nmstate provides both running and in-config routes via `libnmstate.show()`.
+Example:
+
+```python
+{
+    Route.KEY: {
+        Route.CONFIG: [
+            {
+                Route.TABLE_ID: 0,
+                Route.DESTINATION: "0.0.0.0/0",
+                Route.NEXT_HOP_INTERFACE: "eth1",
+                Route.NEXT_HOP_ADDRESS: "192.0.2.1",
+                Route.METRIC: -1
+            }
+        ],
+        Route.RUNNING: [
+            {
+                Route.TABLE_ID: 254,
+                Route.DESTINATION: "192.0.2.0/24",
+                Route.NEXT_HOP_INTERFACE: "eth1",
+                Route.NEXT_HOP_ADDRESS: "",
+                Route.METRIC: 100
+            },
+    }
+}
+```
+
+The running routes contains route entires in kernel, it might contains:
```

Comment:
The running routes contains route **entries** in kernel, **which* might contains